libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalProgressObserver.kt

```diff
@@ -52,6 +53,19 @@ internal class ArrivalProgressObserver(
         val routeLegProgress = routeProgress.currentLegProgress
             ?: return
 
+        when (routeProgress.currentState) {
+            RouteProgressState.LOCATION_TRACKING,
+            RouteProgressState.ROUTE_COMPLETE -> {
+                // continue
+            }
+            RouteProgressState.ROUTE_INITIALIZED,
+            RouteProgressState.OFF_ROUTE,
+            RouteProgressState.ROUTE_UNCERTAIN,
+            RouteProgressState.LOCATION_STALE -> {
```

Comment:
add ROUTE_INVALID?

Reply:
I didn't add it since I'm hoping to remove it entirely https://github.com/mapbox/mapbox-navigation-android/pull/4391. That said, that other PR might need more discussion so I'll update it here, for now, to resolve the immediate issues while we keep iterating.